onnxruntime/core/providers/qnn/builder/qnn_model.h

```diff
@@ -126,9 +134,12 @@ class QnnModel {
   std::vector<std::string> output_names_;
   std::unordered_map<std::string, OnnxTensorInfo> inputs_info_;
   std::unordered_map<std::string, OnnxTensorInfo> outputs_info_;
-  std::vector<Qnn_Tensor_t> qnn_inputs_;
-  std::vector<Qnn_Tensor_t> qnn_outputs_;
+  std::vector<QnnTensorInfo> qnn_input_infos_;
```

Comment:
Note: `Qnn_Tensor_t` contains a pointer to input (or output data). These pointers are necessarily reset on every inference to point to the new input, and then passed to QNN's graphExecute API. Because these are stored as state, multiple threads were overwriting these pointers.

The solution is to not store `Qnn_Tensor_t` objects. Instead, we just store information about the inputs/outputs and then construct a vector of `Qnn_Tensor_t` objects on demand.